### PR TITLE
Switch arcade.math.lerp to use a Protocol bound to a TypeVar

### PR DESCRIPTION
**Pull request by pushfoo**
**TL;DR:** Add `HasAddSubMul` protocol and update `Lerp` functions to return `Vec*` where appropriate

### Changes

* Add `HasAddSubMul` protocol type + `L` TypeVar in `arcade.math`
* Annotate `arcade.math.lerp` cleanly to allow arbitarary lerpables
* Make `lerp_2d` return `Vec2`
* Make `lerp_3d` return `Vec3`
* Docstring updates
* Cross-references for `lerp_angle`

### Follow-up work

- [ ] Should we have `lerp` check for an `object.lerp` method?
- [ ] How do we integrate that into type annotations?

One option for the latter is something like this:
```python
# Assume HasAddSubMul is the same as now

class HasLerp(Protocol[_T, _R]):
    def lerp(other: _T) -> _R:
```
- [ ] Weigh upstreaming the positional slash for maximum fidelity to math built-ins?

### CODE REVIEW

**Comment by pushfoo** [comment]
As an update to this, it looks like only pyglet vector objects have the `lerp` method. The matrices do not.

**Review by DragonMoffon — APPROVED** [review]
